error: static assertion failed: result type must be constructible from input type when building with `-DQT6=ON`

### Bug Description

I cannot build mixxx with `-DQT6=ON` due to this error:

```
In file included from /usr/include/c++/12.2.0/memory:66,
                 from /usr/include/qt6/QtCore/qcontainertools_impl.h:19,
                 from /usr/include/qt6/QtCore/qarraydataops.h:9,
                 from /usr/include/qt6/QtCore/qarraydatapointer.h:7,
                 from /usr/include/qt6/QtCore/qlist.h:8,
                 from /usr/include/qt6/QtCore/qmap.h:9,
                 from /usr/include/qt6/QtCore/QMap:1,
                 from /home/jan/Projects/mixxx/src/effects/backends/builtin/autopaneffect.h:3,
                 from /home/jan/Projects/mixxx/src/effects/backends/builtin/autopaneffect.cpp:1:
/usr/include/c++/12.2.0/bits/stl_uninitialized.h: In instantiation of ‘constexpr bool std::__check_constructible() [with _ValueType = unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> >; _Tp = const unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> >&]’:
/usr/include/c++/12.2.0/bits/stl_uninitialized.h:182:4:   required from ‘_ForwardIterator std::uninitialized_copy(_InputIterator, _InputIterator, _ForwardIterator) [with _InputIterator = __gnu_cxx::__normal_iterator<const unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> >*, vector<unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> >, allocator<unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> > > > >; _ForwardIterator = unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> >*]’
/usr/include/c++/12.2.0/bits/stl_uninitialized.h:372:37:   required from ‘constexpr _ForwardIterator std::__uninitialized_copy_a(_InputIterator, _InputIterator, _ForwardIterator, allocator<_Tp>&) [with _InputIterator = __gnu_cxx::__normal_iterator<const unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> >*, vector<unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> >, allocator<unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> > > > >; _ForwardIterator = unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> >*; _Tp = unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> >]’
/usr/include/c++/12.2.0/bits/stl_vector.h:601:31:   required from ‘constexpr std::vector<_Tp, _Alloc>::vector(const std::vector<_Tp, _Alloc>&) [with _Tp = std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> >; _Alloc = std::allocator<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> > >]’
/usr/include/qt6/QtCore/qvarlengtharray.h:754:17:   required from ‘void QVLABase<T>::reallocate_impl(qsizetype, void*, qsizetype, qsizetype, const T*) [with T = std::vector<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> >, std::allocator<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> > > >; qsizetype = long long int]’
/usr/include/qt6/QtCore/qvarlengtharray.h:211:7:   required from ‘void QVLABase<T>::resize_impl(qsizetype, void*, qsizetype, const T*) [with T = std::vector<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> >, std::allocator<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> > > >; qsizetype = long long int]’
/usr/include/qt6/QtCore/qvarlengtharray.h:376:50:   required from ‘void QVarLengthArray<T, Prealloc>::resize(qsizetype) [with T = std::vector<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> >, std::allocator<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> > > >; long long int Prealloc = 256; qsizetype = long long int]’
/home/jan/Projects/mixxx/src/engine/channelhandle.h:245:30:   required from ‘void ChannelHandleMap<T>::maybeExpand(int) [with T = std::vector<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> >, std::allocator<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> > > >]’
/home/jan/Projects/mixxx/src/engine/channelhandle.h:208:9:   required from ‘T& ChannelHandleMap<T>::operator[](const ChannelHandle&) [with T = std::vector<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> >, std::allocator<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> > > >]’
/home/jan/Projects/mixxx/src/effects/backends/effectprocessor.h:203:57:   required from ‘void EffectProcessorImpl<EffectSpecificState>::initializeInputChannel(ChannelHandle, const mixxx::EngineParameters&) [with EffectSpecificState = AutoPanGroupState]’
/home/jan/Projects/mixxx/src/effects/backends/effectprocessor.h:183:10:   required from here
/usr/include/c++/12.2.0/bits/stl_uninitialized.h:90:56: error: static assertion failed: result type must be constructible from input type
   90 |       static_assert(is_constructible<_ValueType, _Tp>::value,
      |                                                        ^~~~~
/usr/include/c++/12.2.0/bits/stl_uninitialized.h:90:56: note: ‘std::integral_constant<bool, false>::value’ evaluates to false
In file included from /usr/include/c++/12.2.0/variant:45,
                 from /usr/include/qt6/QtCore/qtypeinfo.h:7,
                 from /usr/include/qt6/QtCore/qglobal.h:1397,
                 from /usr/include/qt6/QtCore/qiterator.h:7,
                 from /usr/include/qt6/QtCore/qmap.h:8:
/usr/include/c++/12.2.0/bits/stl_construct.h: In instantiation of ‘constexpr void std::_Construct(_Tp*, _Args&& ...) [with _Tp = unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> >; _Args = {const unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> >&}]’:
/usr/include/c++/12.2.0/bits/stl_uninitialized.h:120:21:   required from ‘constexpr _ForwardIterator std::__do_uninit_copy(_InputIterator, _InputIterator, _ForwardIterator) [with _InputIterator = __gnu_cxx::__normal_iterator<const unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> >*, vector<unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> >, allocator<unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> > > > >; _ForwardIterator = unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> >*]’
/usr/include/c++/12.2.0/bits/stl_uninitialized.h:370:30:   required from ‘constexpr _ForwardIterator std::__uninitialized_copy_a(_InputIterator, _InputIterator, _ForwardIterator, allocator<_Tp>&) [with _InputIterator = __gnu_cxx::__normal_iterator<const unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> >*, vector<unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> >, allocator<unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> > > > >; _ForwardIterator = unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> >*; _Tp = unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> >]’
/usr/include/c++/12.2.0/bits/stl_vector.h:601:31:   required from ‘constexpr std::vector<_Tp, _Alloc>::vector(const std::vector<_Tp, _Alloc>&) [with _Tp = std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> >; _Alloc = std::allocator<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> > >]’
/usr/include/qt6/QtCore/qvarlengtharray.h:754:17:   required from ‘void QVLABase<T>::reallocate_impl(qsizetype, void*, qsizetype, qsizetype, const T*) [with T = std::vector<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> >, std::allocator<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> > > >; qsizetype = long long int]’
/usr/include/qt6/QtCore/qvarlengtharray.h:211:7:   required from ‘void QVLABase<T>::resize_impl(qsizetype, void*, qsizetype, const T*) [with T = std::vector<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> >, std::allocator<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> > > >; qsizetype = long long int]’
/usr/include/qt6/QtCore/qvarlengtharray.h:376:50:   required from ‘void QVarLengthArray<T, Prealloc>::resize(qsizetype) [with T = std::vector<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> >, std::allocator<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> > > >; long long int Prealloc = 256; qsizetype = long long int]’
/home/jan/Projects/mixxx/src/engine/channelhandle.h:245:30:   required from ‘void ChannelHandleMap<T>::maybeExpand(int) [with T = std::vector<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> >, std::allocator<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> > > >]’
/home/jan/Projects/mixxx/src/engine/channelhandle.h:208:9:   required from ‘T& ChannelHandleMap<T>::operator[](const ChannelHandle&) [with T = std::vector<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> >, std::allocator<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> > > >]’
/home/jan/Projects/mixxx/src/effects/backends/effectprocessor.h:203:57:   required from ‘void EffectProcessorImpl<EffectSpecificState>::initializeInputChannel(ChannelHandle, const mixxx::EngineParameters&) [with EffectSpecificState = AutoPanGroupState]’
/home/jan/Projects/mixxx/src/effects/backends/effectprocessor.h:183:10:   required from here
/usr/include/c++/12.2.0/bits/stl_construct.h:115:28: error: no matching function for call to ‘construct_at(std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> >*&, const std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> >&)’
  115 |           std::construct_at(__p, std::forward<_Args>(__args)...);
      |           ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/12.2.0/bits/stl_construct.h:94:5: note: candidate: ‘template<class _Tp, class ... _Args> constexpr decltype (::new(void*(0)) _Tp) std::construct_at(_Tp*, _Args&& ...)’
   94 |     construct_at(_Tp* __location, _Args&&... __args)
      |     ^~~~~~~~~~~~
/usr/include/c++/12.2.0/bits/stl_construct.h:94:5: note:   template argument deduction/substitution failed:
/usr/include/c++/12.2.0/bits/stl_construct.h: In substitution of ‘template<class _Tp, class ... _Args> constexpr decltype (::new(void*(0)) _Tp) std::construct_at(_Tp*, _Args&& ...) [with _Tp = std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> >; _Args = {const std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> >&}]’:
/usr/include/c++/12.2.0/bits/stl_construct.h:115:21:   required from ‘constexpr void std::_Construct(_Tp*, _Args&& ...) [with _Tp = unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> >; _Args = {const unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> >&}]’
/usr/include/c++/12.2.0/bits/stl_uninitialized.h:120:21:   required from ‘constexpr _ForwardIterator std::__do_uninit_copy(_InputIterator, _InputIterator, _ForwardIterator) [with _InputIterator = __gnu_cxx::__normal_iterator<const unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> >*, vector<unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> >, allocator<unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> > > > >; _ForwardIterator = unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> >*]’
/usr/include/c++/12.2.0/bits/stl_uninitialized.h:370:30:   required from ‘constexpr _ForwardIterator std::__uninitialized_copy_a(_InputIterator, _InputIterator, _ForwardIterator, allocator<_Tp>&) [with _InputIterator = __gnu_cxx::__normal_iterator<const unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> >*, vector<unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> >, allocator<unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> > > > >; _ForwardIterator = unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> >*; _Tp = unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> >]’
/usr/include/c++/12.2.0/bits/stl_vector.h:601:31:   required from ‘constexpr std::vector<_Tp, _Alloc>::vector(const std::vector<_Tp, _Alloc>&) [with _Tp = std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> >; _Alloc = std::allocator<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> > >]’
/usr/include/qt6/QtCore/qvarlengtharray.h:754:17:   required from ‘void QVLABase<T>::reallocate_impl(qsizetype, void*, qsizetype, qsizetype, const T*) [with T = std::vector<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> >, std::allocator<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> > > >; qsizetype = long long int]’
/usr/include/qt6/QtCore/qvarlengtharray.h:211:7:   required from ‘void QVLABase<T>::resize_impl(qsizetype, void*, qsizetype, const T*) [with T = std::vector<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> >, std::allocator<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> > > >; qsizetype = long long int]’
/usr/include/qt6/QtCore/qvarlengtharray.h:376:50:   required from ‘void QVarLengthArray<T, Prealloc>::resize(qsizetype) [with T = std::vector<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> >, std::allocator<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> > > >; long long int Prealloc = 256; qsizetype = long long int]’
/home/jan/Projects/mixxx/src/engine/channelhandle.h:245:30:   required from ‘void ChannelHandleMap<T>::maybeExpand(int) [with T = std::vector<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> >, std::allocator<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> > > >]’
/home/jan/Projects/mixxx/src/engine/channelhandle.h:208:9:   required from ‘T& ChannelHandleMap<T>::operator[](const ChannelHandle&) [with T = std::vector<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> >, std::allocator<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> > > >]’
/home/jan/Projects/mixxx/src/effects/backends/effectprocessor.h:203:57:   required from ‘void EffectProcessorImpl<EffectSpecificState>::initializeInputChannel(ChannelHandle, const mixxx::EngineParameters&) [with EffectSpecificState = AutoPanGroupState]’
/home/jan/Projects/mixxx/src/effects/backends/effectprocessor.h:183:10:   required from here
/usr/include/c++/12.2.0/bits/stl_construct.h:96:17: error: use of deleted function ‘std::unique_ptr<_Tp, _Dp>::unique_ptr(const std::unique_ptr<_Tp, _Dp>&) [with _Tp = AutoPanGroupState; _Dp = std::default_delete<AutoPanGroupState>]’
   96 |     -> decltype(::new((void*)0) _Tp(std::declval<_Args>()...))
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/12.2.0/memory:76:
/usr/include/c++/12.2.0/bits/unique_ptr.h:514:7: note: declared here
  514 |       unique_ptr(const unique_ptr&) = delete;
      |       ^~~~~~~~~~
/usr/include/c++/12.2.0/bits/stl_construct.h: In instantiation of ‘constexpr void std::_Construct(_Tp*, _Args&& ...) [with _Tp = unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> >; _Args = {const unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> >&}]’:
/usr/include/c++/12.2.0/bits/stl_uninitialized.h:120:21:   required from ‘constexpr _ForwardIterator std::__do_uninit_copy(_InputIterator, _InputIterator, _ForwardIterator) [with _InputIterator = __gnu_cxx::__normal_iterator<const unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> >*, vector<unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> >, allocator<unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> > > > >; _ForwardIterator = unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> >*]’
/usr/include/c++/12.2.0/bits/stl_uninitialized.h:370:30:   required from ‘constexpr _ForwardIterator std::__uninitialized_copy_a(_InputIterator, _InputIterator, _ForwardIterator, allocator<_Tp>&) [with _InputIterator = __gnu_cxx::__normal_iterator<const unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> >*, vector<unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> >, allocator<unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> > > > >; _ForwardIterator = unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> >*; _Tp = unique_ptr<AutoPanGroupState, default_delete<AutoPanGroupState> >]’
/usr/include/c++/12.2.0/bits/stl_vector.h:601:31:   required from ‘constexpr std::vector<_Tp, _Alloc>::vector(const std::vector<_Tp, _Alloc>&) [with _Tp = std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> >; _Alloc = std::allocator<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> > >]’
/usr/include/qt6/QtCore/qvarlengtharray.h:754:17:   required from ‘void QVLABase<T>::reallocate_impl(qsizetype, void*, qsizetype, qsizetype, const T*) [with T = std::vector<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> >, std::allocator<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> > > >; qsizetype = long long int]’
/usr/include/qt6/QtCore/qvarlengtharray.h:211:7:   required from ‘void QVLABase<T>::resize_impl(qsizetype, void*, qsizetype, const T*) [with T = std::vector<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> >, std::allocator<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> > > >; qsizetype = long long int]’
/usr/include/qt6/QtCore/qvarlengtharray.h:376:50:   required from ‘void QVarLengthArray<T, Prealloc>::resize(qsizetype) [with T = std::vector<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> >, std::allocator<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> > > >; long long int Prealloc = 256; qsizetype = long long int]’
/home/jan/Projects/mixxx/src/engine/channelhandle.h:245:30:   required from ‘void ChannelHandleMap<T>::maybeExpand(int) [with T = std::vector<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> >, std::allocator<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> > > >]’
/home/jan/Projects/mixxx/src/engine/channelhandle.h:208:9:   required from ‘T& ChannelHandleMap<T>::operator[](const ChannelHandle&) [with T = std::vector<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> >, std::allocator<std::unique_ptr<AutoPanGroupState, std::default_delete<AutoPanGroupState> > > >]’
/home/jan/Projects/mixxx/src/effects/backends/effectprocessor.h:203:57:   required from ‘void EffectProcessorImpl<EffectSpecificState>::initializeInputChannel(ChannelHandle, const mixxx::EngineParameters&) [with EffectSpecificState = AutoPanGroupState]’
/home/jan/Projects/mixxx/src/effects/backends/effectprocessor.h:183:10:   required from here
/usr/include/c++/12.2.0/bits/stl_construct.h:119:7: error: use of deleted function ‘std::unique_ptr<_Tp, _Dp>::unique_ptr(const std::unique_ptr<_Tp, _Dp>&) [with _Tp = AutoPanGroupState; _Dp = std::default_delete<AutoPanGroupState>]’
  119 |       ::new((void*)__p) _Tp(std::forward<_Args>(__args)...);
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/12.2.0/bits/unique_ptr.h:514:7: note: declared here
  514 |       unique_ptr(const unique_ptr&) = delete;
      |       ^~~~~~~~~~
make[2]: *** [CMakeFiles/mixxx-lib.dir/build.make:1462: CMakeFiles/mixxx-lib.dir/src/effects/backends/builtin/autopaneffect.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:298: CMakeFiles/mixxx-lib.dir/all] Error 2
make: *** [Makefile:166: all] Error 2
```

IIUC the issue is that it tries to make a copy of the `unique_ptr` when accessing the `ChannelHandleMap<std::vector<std::unique_ptr<EffectSpecificState>>>`:

```cpp
auto& outputChannelStates = m_channelStateMatrix[inputChannel];
```

### Version

Latest main (86a280b5e03efc0a374226f9fd6c74274be445d6)

### OS

Arch Linux with gcc 12.2.0-1 on AMD64

--- a/src/effects/backends/effectprocessor.h
+++ b/src/effects/backends/effectprocessor.h
@@ -12,6 +12,7 @@
 #include "engine/engine.h"
 #include "util/sample.h"
 #include "util/types.h"
+#include "util/unique_ptr_vector.h"
 
 /// Effects are implemented as two separate classes, an EffectState subclass and
 /// an EffectProcessorImpl subclass. Separating state from the DSP code allows
@@ -246,5 +247,5 @@ class EffectProcessorImpl : public EffectProcessor {
 
   private:
     QSet<ChannelHandleAndGroup> m_registeredOutputChannels;
-    ChannelHandleMap<std::vector<std::unique_ptr<EffectSpecificState>>> m_channelStateMatrix;
+    ChannelHandleMap<unique_ptr_vector<EffectSpecificState>> m_channelStateMatrix;
 };

--- a/src/util/unique_ptr_vector.h
+++ b/src/util/unique_ptr_vector.h
@@ -1,0 +1,21 @@
+#pragma once
+
+#include <memory>
+#include <vector>
+
+// This wrapper class can be used as a workaround for bug in QT 6.4
+// https://bugreports.qt.io/browse/QTBUG-109745 that prevents to use
+// std::vector<std::unique_ptr<T>> in QVarLengthArray
+// The only purpose of this class is to delete the copy constructor from
+// std::vector<std::unique_ptr<T>> and make std::is_copy_constructible_v<> false
+// std::vector<std::unique_ptr<T>> cannot be copied even if it is empty due to a
+// static_assert in stl_uninitialized.h
+template<typename T>
+class unique_ptr_vector : public std::vector<std::unique_ptr<T>> {
+    using std::vector<std::unique_ptr<T>>::vector;
+
+  public:
+    unique_ptr_vector(const unique_ptr_vector&) = delete;
+    unique_ptr_vector(unique_ptr_vector&&) noexcept = default;
+    unique_ptr_vector& operator=(const unique_ptr_vector&);
+};

